### ISSUE TITLE
feat: implement global per-domain request scheduler with rate limiting and circuit breakers

### ISSUE THREAD

**Issue by berntpopp** [berntpopp/kidney-genetics-db#129]
## Problem

The data pipeline makes hundreds of requests to 5+ external APIs (JAX HPO, NCBI PubTator, GenCC, ClinGen, PanelApp) without global rate limiting. Each source manages its own retry logic independently, leading to:

- **429 retry storms**: When the JAX API rate-limits, hundreds of concurrent requests all retry independently, causing 100% CPU spin
- **No per-domain awareness**: Requests to the same domain from different code paths don't share rate state
- **Per-request retry instead of queue-level backoff**: A 429 should pause ALL requests to that domain, not just the one that received it

### Evidence
- HPO source: 600+ individual API calls to `ontology.jax.org`, batch_size=10, 0.1s delay → immediate 429 flood → 100% CPU
- `raise_for_status()` was missing so 429s were silently swallowed (fixed in 7cbb056, but root cause remains)

## Proposed Solution

Implement a **rate-limited httpx transport layer** with per-domain configuration:

### Architecture

```
httpx.AsyncClient(transport=RateLimitedTransport(registry))
    └── RateLimitedTransport
        └── DomainRateLimiterRegistry
            ├── hpo.jax.org:        2 req/s, max 2 concurrent, circuit breaker
            ├── ncbi.nlm.nih.gov:   3 req/s, max 3 concurrent, circuit breaker
            ├── thegencc.org:       10 req/s, max 5 concurrent
            ├── clinicalgenome.org: 5 req/s, max 3 concurrent
            └── panelapp...:        5 req/s, max 3 concurrent
```

### Key components

1. **`DomainRateLimiterRegistry`**: Per-domain rate limiters (leaky bucket via `aiolimiter`) + concurrency semaphores
2. **`RateLimitedTransport`**: Custom `httpx.AsyncBaseTransport` wrapper that acquires rate limit + semaphore before each request
3. **Queue-level 429 handling**: When a 429 is received, pause ALL workers for that domain using the `Retry-After` header value
4. **Circuit breakers per domain**: After N consecutive failures, stop sending requests for a recovery period
5. **Adaptive backoff with full jitter**: Prevent thundering herd on retry

### Benefits of transport-layer approach

- **Transparent**: All existing `client.get()`/`client.post()` calls are automatically governed — minimal code changes
- **Global**: All requests to the same domain share rate state, regardless of which source/module initiates them
- **No retry storms**: 429 pauses the domain, not individual requests

### Dependencies

```
aiolimiter>=1.1.0   # Leaky bucket rate limiter (zero deps, asyncio-native)
```

`asyncio.Semaphore` (stdlib) for concurrency. No other new deps needed.

### Configuration

Rate limits should be configurable per domain in `config/annotations.yaml` or similar:

```yaml
rate_limits:
  ontology.jax.org:
    max_rate: 2          # requests per second
    max_concurrent: 2
    circuit_breaker:
      failure_threshold: 10
      recovery_timeout: 60
  www.ncbi.nlm.nih.gov:
    max_rate: 3
    max_concurrent: 3
```

### References

- [AWS Architecture Blog: Exponential Backoff and Jitter](https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/)
- [httpx Custom Transports](https://www.python-httpx.org/advanced/transports/)
- [aiolimiter docs](https://aiolimiter.readthedocs.io/)

Supersedes #126.
